### PR TITLE
Add support for Elixir (need help with multiple node error)

### PR DESCRIPTION
**Pull request by dimitarvp**
Please let me know if I have missed any of the bucket list items needed to add support for a new language.

Still, this PR is not ready to merge for the following reasons:

1. The `tree-sitter-elixir` dependency is only available through GIT and not via crates.io because the crate has not been updated in a while and it points to a now non-existing repo. I've contacted the maintainers and I hope they will repair the crate soon as the GIT repo and code exist but are at another location (see `Cargo.toml`).
2. The Elixir tests are very bare because e.g. doing matching like this:

```rust
  test_match("$A != nil", "a != nil");
```

Yields this error:

```plain
called `Result::unwrap()` on an `Err` value: MultipleNode("$A != nil")
```

I admit I know nothing about TreeSitter except what it is doing (and even that I don't know very well), plus I don't know the codebase of this project well enough. Anyone willing to provide guidance on how can I troubleshoot this error and make the tests less bare and a bit more convincing?

Finally, compiling `ast-grep` locally with the new Elixir support and doing `$FORKED_ASTGREP_PATH/target/release/sg -p '$A != nil' --lang elixir` inside an Elixir project directory yields the multiple node error again, reducing the usefulness of the tool.

Any advice will be appreciated, thank you!

### CODE REVIEW

**Comment by HerringtonDarkholme** [comment]
Hi, thanks for trying out ast-grep.

It looks like Elixir tree-sitter parser is not well written. The multiple node error

```
$A != nil
```

means the pattern is not parsed as one AST node. This is probably caused by parsing errors. If you are already developing ast-grep's Rust code. You can print the node's sexp in pattern.rs

Plus, tree-sitter-elixir is not released on crates.io. I still think there is a lot to improve the quality of elixir parser. See https://ast-grep.github.io/contributing/add-lang.html#requirements-and-constraints


Alternatively, you can also try using elixir parsr as a [custom language](https://ast-grep.github.io/advanced/custom-language.html)

**Comment by dimitarvp** [comment]
Thanks for responding! Appreciate it.

I'll await for a response from the authors of `tree-sitter-elixir` and see what we can do.

I'll also try your suggestion about printing the node(s) and see if I can do something productive with the output.

Let's keep this PR open, I'll do my best to resolve the outstanding issues ASAP and then check if we can merge it.

File: crates/language/src/parsers.rs [hunk 1]
```diff
@@ -23,6 +23,9 @@ mod parser_implementation {
   pub fn language_dart() -> TSLanguage {
     tree_sitter_dart::language().into()
   }
+  pub fn language_elixir() -> TSLanguage {
+    tree_sitter_elixir::language().into()
```
Comment (HerringtonDarkholme) [inline crates/language/src/parsers.rs:27]: You also need to add language_elixir in `#[cfg(not(feature = "builtin-parser"))]`

See https://ast-grep.github.io/contributing/add-lang.html#add-to-ast-grep-core

Comment (dimitarvp) [reply crates/language/src/parsers.rs:27]: Just did, thank you.

**Comment by HerringtonDarkholme** [comment]
Looks nice. Thanks

**Pull request merged**